src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,96 @@
 package core.basesyntax;
 
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void createNewArray() {
+        int newSize = array.length + array.length / 2;
+        T[] newArray = (T[]) new Object[newSize];
+        System.arraycopy(array, 0, newArray, 0, newSize);
```

Comment:
The `System.arraycopy` destination should be `array` not `newArray`, and the length of elements to be copied should be `size` not `newSize`. This is a bug that will cause incorrect array copying.